### PR TITLE
Major overhaul of benchmarking

### PR DESCRIPTION
**Pull request by mseeger**
*Issue #, if available:*

*Description of changes:*
This is a major clean-up of benchmarking. In fact, things did not work properly before. This contains several fixes.

- Reduce arguments in MethodArguments, in favor of scheduler_kwargs, which collect method-specific args
- Make sure that scheduler_kwargs is properly modified by extra arguments. This requires recursive merging of dictionaries. The current code does not do the right thing
- Simplify things a lot for people creating new experiment in nursery. There are wrappers now for all baselines, you can just pick what you want, not worrying about how arguments are put together based on command line args. Look at any module in nursery, how simple this is now
- Updated benchmarking tutorial

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by wesk** [comment]
`Integ Tests` are currently configured to run with each PR, whereas the `End-to-end Tests` that have AWS credentials only run manually on-demand and before each release.

File: benchmarking/commons/default_baselines.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        default_arguments(method_arguments, da_input),
+        kwargs,
+        stop_keys=["config_space"],
+    )
+    return result
+
+
+def RandomSearch(method_arguments: MethodArguments, **kwargs):
+    return _RandomSearch(**_baseline_kwargs(method_arguments, kwargs))
+
```
Comment (jgolebiowski) [inline benchmarking/commons/default_baselines.py:63]: @this and all other definitions: Thats really cool and makes defining new benchmarks much easier!

**Review by jgolebiowski — APPROVED** [review]
Thats a great change. I am not sure about the defined type hints (see comment) but that should not block merging if this is urgent for further changes.

File: benchmarking/commons/hpo_main_common.py [hunk 2]
```diff
@@ -13,18 +13,31 @@
...
+
+MapExtraArgsType = Callable[[Any, str, DictStrKey], DictStrKey]
+
+
+ExtraArgsType = List[DictStrKey]
+
+
+PostProcessingType = Callable[[Tuner], Any]
+
+
```
[the remark this answers is not in the document]
Comment (mseeger) [reply benchmarking/commons/hpo_main_common.py:38]: You are right for DictStrKey, I can remove this.

But MapExtraArgsType and PostProcessingType gets pretty complex.

File: benchmarking/commons/default_baselines.py (hunk 1 above)
Comment (mseeger) [reply benchmarking/commons/default_baselines.py:63]: Yes, now you only need to import default_baselines, and then you can use it almost like optimizer/baselines, but all the arguments are done behind the scenes.

**Pull request merged**